File: two_pointers/125_valid_palindrome.py
def is_palindrome(s: str) -> bool:
    l = 0
    r = len(s) - 1

    while l < r:
        left = s[l].lower()
        right = s[r].lower()

        if not left.isalnum():
            l += 1
            continue

        if not right.isalnum():
            r -= 1
            continue

        if left != right:
            return False

        l += 1
        r -= 1

    return True

File: two_pointers/125_valid_palindrome.go
import "unicode"

func isAlphanumeric(s rune) bool {
	return unicode.IsLetter(s) || unicode.IsDigit(s)
}

func isPalindrome(s string) bool {
	runes := []rune(s)
	l := 0
	r := len(s) - 1

	for l < r {
		left := unicode.ToLower(rune(runes[l]))
		if !isAlphanumeric(unicode.ToLower(left)) {
			l++
			continue
		}

		right := unicode.ToLower(rune(runes[r]))
		if !isAlphanumeric(right) {
			r--
			continue
		}

		if left != right {
			return false
		}

		l++
		r--
	}
	return true
}
